homeassistant/components/stream/hls.py

```diff
@@ -126,10 +269,12 @@ async def handle(self, request, stream, sequence):
         """Return init.mp4."""
         track = stream.add_provider("hls")
         segments = track.get_segment()
-        if not segments:
+        if not segments or not segments[0].first_moof_loc:
```

Comment:
Since we can have an incomplete segment we need to test whether the init is ready or not. We can also test for `segments[0].init` if that's more explicit.